scripts/panel.py

```
@@ -613,11 +698,23 @@ def cmd_run(args):
         die("no API key found (OPENROUTER_API_KEY / AR_API_KEY / AR_KEY_FILE). "
             "For keyless MCP transport use `panel.py prepare` + `panel.py ingest`.", 2)
 
+    # Establish (and persist, on first call) the run's cost ceiling; it stays authoritative for
+    # rebuttal and concurrence too, so the audit shows which cap was actually enforced.
+    cap, cap_src = run_cost_cap(run)
```

Ruff (0.16.1)

[warning] Unpacked variable `cap_src` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)